filters/include/pcl/filters/experimental/functor_filter.h

```diff
@@ -77,22 +75,50 @@ class FunctorFilter : public FilterIndices<PointT> {
    */
   void
   applyFilter(Indices& indices) override
+  {
+    auto filter = [&](auto exec) {
+      applyFilter(exec, indices);
+    };
+
+    executor::enable_exec_on_desc_priority(filter, executor::default_omp_executor{}, executor::default_inline_executor{});
+  }
+
+  template <typename Executor, typename executor::InstanceOfAny<Executor, executor::inline_executor, executor::omp_executor> = 0>
+  void
+  applyFilter (const Executor &exec, Indices& indices)
   {
     indices.clear();
-    indices.reserve(input_->size());
+    indices.reserve(indices_->size());
+    std::vector<std::uint8_t> keep(indices_->size(), 0);
+
     if (extract_removed_indices_) {
       removed_indices_->clear();
-      removed_indices_->reserve(input_->size());
+      removed_indices_->reserve(indices_->size());
     }
 
-    for (const auto index : *indices_) {
-      // functor returns true for points that should be selected
-      if (negative_ != functor_(*input_, index)) {
-        indices.push_back(index);
-      }
-      else if (extract_removed_indices_) {
-        removed_indices_->push_back(index);
+    auto cond_filtering = [&](executor::executor_index_t<Executor> index) {
+      pcl::utils::ignore(index);
+#pragma omp for
+      for (std::ptrdiff_t idx = 0; idx < indices_->size(); ++idx) {
+        if (negative_ != functor_(*input_, idx)) {
```

Comment:
there's a bug here
option A
```suggestion
      for (std::size_t idx = 0; idx < indices_->size(); ++idx) {
        if (negative_ != functor_(*input_, (*indices)[idx])) {
```

option B
```suggestion
      for (uindex_t idx = 0; idx < indices_->size(); ++idx) {
        if (negative_ != functor_(*input_, (*indices)[idx])) {
```

Reply:
We need to add a test for a filter with missing indices to catch such errors